packages/core/src/Stores/client-store.js

```diff
@@ -420,9 +420,10 @@ export default class ClientStore extends BaseStore {
             () => {
                 const { trading_hub } = this.account_settings;
                 const is_traders_hub = !!trading_hub;
-
+                const { preferred_language } = this.account_settings?.preferred_language;
+                const language = preferred_language && preferred_language !== 'EN' ? '' : '';
                 if (!this.is_pre_appstore && window.location.pathname === routes.traders_hub) {
-                    window.location.href = routes.root;
+                    window.location.href = routes.root.replace(/\?{0,1}lang=.{2}$/i, language);
```

Comment:
Because somewhere in the code it overwrites and removes the `lang?` param and it only happens for new singups. This is to check if the language from `preferred_language` exist  and appended at the end for the redirect url.

Reply:
Where is "somewhere" in the code?